foreman/data_refinery_foreman/surveyor/external_source.py

```diff
@@ -28,160 +32,127 @@ class ExternalSourceSurveyor:
         return
 
     @retry(stop_max_attempt_number=3)
-    def queue_downloader_jobs(self, batches: List[Batch]):
-        if len(batches) > 0:
-            downloader_task = self.downloader_task()
+    def queue_downloader_jobs(self, experiment: Experiment):
+
+        # Get all of the undownloaded original files related to this Experiment.
+        relations = ExperimentSampleAssociation.objects.filter(experiment=experiment)
+        samples = Sample.objects.filter(id__in=relations.values('sample_id'))
+        files_to_download = OriginalFile.objects.filter(sample__in=samples.values('pk'), is_downloaded=False)
+
+        downloaded_urls = []
+        for original_file in files_to_download:
+
+            # We don't need to create multiple downloaders for the same file.
+            if original_file.source_url in downloaded_urls:
+              continue
+
+            # There is already a downloader job associated with this file.
+            old_assocs = DownloaderJobOriginalFileAssociation.objects.filter(original_file__source_url=original_file.source_url)
+            if len(old_assocs) > 0:
+              logger.debug("We found an existing DownloaderJob for this file/url.", original_file_id=original_file.id)
+              continue
 
             with transaction.atomic():
-                downloader_job = DownloaderJob.create_job_and_relationships(
-                    batches=batches, downloader_task=downloader_task.value)
 
-            logger.info("Queuing downloader job.",
+              downloader_job = DownloaderJob()
+              downloader_job.downloader_task = self.downloader_task()
+              downloader_job.accession_code = experiment.accession_code
+              downloader_job.save()
+
+              asoc = DownloaderJobOriginalFileAssociation()
+              asoc.downloader_job = downloader_job
+              asoc.original_file = original_file
+              asoc.save()
+
+              downloaded_urls.append(original_file.source_url)
+
+            try:
+                logger.info("Queuing downloader job.",
                         survey_job=self.survey_job.id,
                         downloader_job=downloader_job.id)
-            try:
-                send_job(downloader_task, downloader_job.id)
+                send_job(downloader_job.downloader_task, downloader_job.id)
             except:
                 # If the task doesn't get sent we don't want the
                 # downloader_job to be left floating
+                logger.info("Failed to enqueue downloader job.",
+                        survey_job=self.survey_job.id,
+                        downloader_job=downloader_job.id)
                 downloader_job.delete()
                 raise
-        else:
-            logger.info("Survey job found no new Batches.",
-                        survey_job=self.survey_job.id)
+
+    @retry(stop_max_attempt_number=3)
+    def queue_downloader_job_for_original_files(self, original_files: List[OriginalFile]):
```

Comment:
Can you add a docstring to this function and `queue_downloader_jobs()` clarifying that one queues a different downloader job for each file and that the other queues one downloader job for all files? I figured this out the last time I went through the code but it took me a while to re-figure it out this pass.